src/utils/SocketClient.hxx

```diff
@@ -71,17 +84,21 @@ public:
      *                      connecting on error.
      * @param timeout_seconds time in seconds that the connect is supposed to
      *                        timeout and look for a possible shutdown.
+     * @param status callback method for status as the connection attempt
+     *               progresses
      */
     SocketClient(Service *service, const char *mdns, const char *host,
                  uint16_t port,
                  std::function<void(int, struct addrinfo *, Notifiable*)>callback,
-                 uint8_t retry_seconds = 5, uint8_t timeout_seconds = 255)
+                 uint8_t retry_seconds = 5, uint8_t timeout_seconds = 255,
+                 std::function<void(SocketStatus)>status = nullptr)
```

Comment:
please move the status callback before the integer arguments. Consider renaming it to status_callback.